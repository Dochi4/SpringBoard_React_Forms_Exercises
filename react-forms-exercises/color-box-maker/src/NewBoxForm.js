import React, { useState } from "react";

const NewBoxForm = ({ addBox }) => {
  const initialState = {
    width: "",
    height: "",
    background: "",
  };

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addBox({
      ...formData,
      width: `${formData.width}px`,
      height: `${formData.height}px`,
    });
    setFormData(initialState);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="width">Width (px)</label>
      <input
        type="number"
        id="width"
        name="width"
        value={formData.width}
        onChange={handleChange}
        placeholder="Add Width"
      />

      <label htmlFor="height">Height (px)</label>
      <input
        type="number"
        id="height"
        name="height"
        value={formData.height}
        onChange={handleChange}
        placeholder="Add Height"
      />

      <label htmlFor="background">Background Color</label>
      <input
        type="text"
        id="background"
        name="background"
        value={formData.background}
        onChange={handleChange}
        placeholder="Add Color"
      />

      <button>Add Box</button>
    </form>
  );
};

export default NewBoxForm;
