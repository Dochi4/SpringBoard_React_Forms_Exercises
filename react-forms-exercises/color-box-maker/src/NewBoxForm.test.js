import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";
import TEST_BOX from "./_testCommon";

it("should render without crashing", function () {
  render(<NewBoxForm addBox={TEST_BOX} />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<NewBoxForm addBox={TEST_BOX} />);
  expect(asFragment()).toMatchSnapshot();
});

it("should change the width input", function () {
  render(<NewBoxForm addBox={TEST_BOX} />);

  const widthInput = screen.getByLabelText(/width/i);
  fireEvent.change(widthInput, { target: { value: "200" } });
  expect(widthInput.value).toBe("200");
});
it("should change the height input", function () {
  render(<NewBoxForm addBox={TEST_BOX} />);

  const heightInput = screen.getByLabelText(/height/i);
  fireEvent.change(heightInput, { target: { value: "200" } });
  expect(heightInput.value).toBe("200");
});

it("should change the background input", function () {
  render(<NewBoxForm addBox={TEST_BOX} />);

  const backgroundInput = screen.getByLabelText(/background/i);
  fireEvent.change(backgroundInput, { target: { value: "red" } });
  expect(backgroundInput.value).toBe("red");
});
