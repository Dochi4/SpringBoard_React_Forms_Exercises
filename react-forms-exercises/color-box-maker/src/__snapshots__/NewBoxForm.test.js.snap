// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width (px)
    </label>
    <input
      id="width"
      name="width"
      placeholder="Add Width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height (px)
    </label>
    <input
      id="height"
      name="height"
      placeholder="Add Height"
      type="number"
      value=""
    />
    <label
      for="background"
    >
      Background Color
    </label>
    <input
      id="background"
      name="background"
      placeholder="Add Color"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
